### PR TITLE
XmlSuite.toXml() missing suite time-out property

### LINKED ISSUE #778: XmlSuite toXml() does NOT add the suite time-out property.
```
    if (get_liveMode().equalsIgnoreCase("true")) {
        suite.setTimeOut(getProject().getProperty(
                "suite.live.windows.timeout"));
    } else {
        suite.setTimeOut(getProject().getProperty("suite.windows.timeout"));
    }
    log("Timeout set to " + suite.getTimeOut());
    if (suite.getTimeOut().isEmpty()) {
        suite.setTimeOut("60000");
    }
            ....
            suite.toXml() doesn't show time-out=60000 as one would expect. 
```

/**
- @return A String representation of this XML suite.
  */
  public String toXml() {
    XMLStringBuffer xsb = new XMLStringBuffer();
    xsb.setDocType("suite SYSTEM \"" + Parser.TESTNG_DTD_URL + '\"');
    Properties p = new Properties();
  ....

### PR DESCRIPTION
**Pull request by krmahadevan**
Closes #778

Fixes #778 .

### Did you remember to?

- [X] Add test case(s)
- [X] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

**Comment by krmahadevan** [comment]
ping @juherr - Please help take a look.

File: src/test/java/test/SimpleBaseTest.java [hunk 1]
```diff
@@ -345,7 +346,21 @@ protected static File createDirInTempDir(String dir) {
...
     List<Integer> resultLineNumbers = new ArrayList<>();
-    try (BufferedReader fr = new BufferedReader(new FileReader(fileName))) {
+    Reader reader = null;
+    try {
+      reader = new FileReader(fileName);
+      resultLineNumbers =  grep(reader, regexp, resultLines);
+    }catch (IOException e) {
+      e.printStackTrace();
+    } finally {
+      closeQuietly(reader);
```
Comment (juherr) [inline src/test/java/test/SimpleBaseTest.java]: We use Java7, so you should use the try-with-resources feature.

Comment (juherr) [reply src/test/java/test/SimpleBaseTest.java]: If it is intentional, just add a comment which explains why

Comment (krmahadevan) [reply src/test/java/test/SimpleBaseTest.java]: I will fix this. The `closeQuietly()` call should have been happening in the `BufferedReader`.

Comment (krmahadevan) [reply src/test/java/test/SimpleBaseTest.java]: @juherr - I have amended the commit and addressed this.

File: src/test/java/test/SimpleBaseTest.java [hunk 2]
```diff
@@ -345,7 +347,19 @@ protected static File createDirInTempDir(String dir) {
    */
   protected static List<Integer> grep(File fileName, String regexp, List<String> resultLines) {
     List<Integer> resultLineNumbers = new ArrayList<>();
-    try (BufferedReader fr = new BufferedReader(new FileReader(fileName))) {
+    Reader reader;
+    try {
+      reader = new FileReader(fileName);
+      resultLineNumbers =  grep(reader, regexp, resultLines);
+    }catch (FileNotFoundException e) {
```
Comment (juherr) [inline src/test/java/test/SimpleBaseTest.java]: Typo: space missing

File: src/test/java/test/SimpleBaseTest.java [hunk 3]
```diff
@@ -345,7 +347,19 @@ protected static File createDirInTempDir(String dir) {
    */
   protected static List<Integer> grep(File fileName, String regexp, List<String> resultLines) {
     List<Integer> resultLineNumbers = new ArrayList<>();
-    try (BufferedReader fr = new BufferedReader(new FileReader(fileName))) {
+    Reader reader;
+    try {
+      reader = new FileReader(fileName);
```
Comment (juherr) [inline src/test/java/test/SimpleBaseTest.java]: The reader should be located in the try even if we catch FileNotFoundException

**Review by juherr — CHANGES_REQUESTED** [review]

File: src/test/java/test/SimpleBaseTest.java (hunk 2 above)
Comment (krmahadevan) [reply src/test/java/test/SimpleBaseTest.java]: Fixed, amended commit and pushed.

File: src/test/java/test/SimpleBaseTest.java (hunk 3 above)
Comment (krmahadevan) [reply src/test/java/test/SimpleBaseTest.java]: Fixed, amended commit and pushed.

**Review by juherr — APPROVED** [review]

**Pull request merged**